### PR TITLE
feat/SRS-373 Implement parcel description front end for internal users

### PR DESCRIPTION
**Pull request by bolny**
# SRS-373 External - Parcel Description Page Purchased - Site Details

This update implements the parcel descriptions navigation component for the site details front end.

Due to the size and complexity of this feature I'll be making a number of pull requests:

- [x] Database Migration
- [x] Backend functionality for internal users
- [x] Shared component updates
- [x] Front end functionality for internal users
- [ ] External user (snapshot) support for both front end and back end.

### CODE REVIEW

File: frontend/site-registry/src/app/features/details/navigation/NavigationPillsConfig.tsx [hunk 1]
```diff
@@ -19,16 +20,18 @@ export const navItems: string[] = [
...
-  <Summary />,
-  <Notations />,
-  <Participants />,
-  <Documents />,
-  <Associate />,
-  <LandUses />,
-  <></>,
-  <Disclosure />,
-];
+export function navComponents(siteId: number): JSX.Element[] {
```
Comment (bolny) [inline frontend/site-registry/src/app/features/details/navigation/NavigationPillsConfig.tsx]: I'm not 100% certain that this is the best way to pass the currently viewed site id into my component. Let me know if this is alright or if there is another, better, option that isn't immediately apparent to me.

Comment (midhun-aot) [reply frontend/site-registry/src/app/features/details/navigation/NavigationPillsConfig.tsx]: Site ID  can be fetched from the URL.  like this   const { id } = useParams();

File: frontend/site-registry/src/app/features/details/parcelDescriptions/getParcelDescriptions.ts [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+  page: number,
+  pageSize: number,
+  searchParam: string,
+  sortBy: string,
+  sortByDir: string,
+) {
+  const axios = getAxiosInstance();
+  const response = await axios.post(GRAPHQL, {
+    query: print(gql`
+      query getParcelDescriptionBySiteId(
```
Comment (midhun-aot) [inline frontend/site-registry/src/app/features/details/parcelDescriptions/getParcelDescriptions.ts]: Better to move graphql query to separate file for more readability

File: frontend/site-registry/src/app/features/details/parcelDescriptions/getParcelDescriptions.ts [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+      siteId: siteId,
+      page: page,
+      pageSize: pageSize,
+      searchParam: searchParam,
+      sortBy: sortBy,
+      sortByDir: sortByDir,
+    },
+  });
+
+  let rawData = response?.data?.data?.getParcelDescriptionsBySiteId;
```
Comment (midhun-aot) [inline frontend/site-registry/src/app/features/details/parcelDescriptions/getParcelDescriptions.ts]: We need to verify http response code before accessing content. Please refer to existing implementations such as folio or cart

File: frontend/site-registry/src/app/features/details/parcelDescriptions/getParcelDescriptions.ts [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+import gql from 'graphql-tag';
+import { GRAPHQL } from '../../../helpers/endpoints';
+import { getAxiosInstance } from '../../../helpers/utility';
+import {
+  IParcelDescriptionDto,
+  IParcelDescriptionResponseDto,
+} from './parcelDescriptionDto';
+import { format } from 'date-fns';
+
+export async function getParcelDescriptions(
```
Comment (midhun-aot) [inline frontend/site-registry/src/app/features/details/parcelDescriptions/getParcelDescriptions.ts]: This should be part of redux & data should be managed in redux . getParcelDescriptions should be implemented using createAsyncThunk

File: frontend/site-registry/src/app/features/details/parcelDescriptions/parcelDescriptions.tsx [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+  const totalResults = parcelDescriptionsState.totalResults;
+  const currentPage = parcelDescriptionsState.currentPage;
+  const resultsPerPage = parcelDescriptionsState.resultsPerPage;
+  const searchParam = parcelDescriptionsState.searchParam;
+  const sortBy = parcelDescriptionsState.sortBy;
+  const sortByDir = parcelDescriptionsState.sortByDir;
+  const sortByInputValue = parcelDescriptionsState.sortByInputValue;
+
+  async function updateParcelDescriptions() {
+    let parcelDescriptions = await getParcelDescriptions(
```
Comment (midhun-aot) [inline frontend/site-registry/src/app/features/details/parcelDescriptions/parcelDescriptions.tsx:47]: If we move this method as a redux action then we can avoid this.

**Review by midhun-aot — CHANGES_REQUESTED** [review]

**Review by midhun-aot — APPROVED** [review]

**Pull request merged**